File: 프로그래머스/2/42842. 카펫/카펫.js
// function solution(brown, yellow) {
//     let [row, col] = [0, 0];
    
//     for (let col = 1; col < 5000; col++) {
//         for (let row = 1; row < 5000; row++) {
//             if ((row * 2) + (col * 2) - 4 == brown && (row - 2) * (col - 2) == yellow) {
//                 return [row, col]
//             }     
//         }  
//     }
// }

function solution(brown, yellow) {
    let [row, col] = [0, 0];
    
    for (let col = 0; col < (brown+yellow); col++) {
        const row = Math.floor((brown + yellow) / col);
        
        if ((row * 2) + (col * 2) - 4 == brown) {
            if ((row - 2) * (col - 2) == yellow) {
                return [row, col]
            }
        }
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
import math

def check(row, col, brown):
    return (row + col - 2) * 2 == brown


def solution(brown, yellow):
    # brown + yellow = row * col
    # brown = (row + col - 2) * 2
    
    for i in range(2, int(math.sqrt(brown + yellow)) + 1):
        if (brown + yellow) % i == 0:
            if check((brown + yellow) // i, i, brown):
                return [(brown + yellow) // i, i]